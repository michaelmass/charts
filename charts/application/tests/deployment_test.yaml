suite: test creating a simple deployment
templates:
  - deployment.yaml
tests:
  - it: should be empty
    set: {}
    asserts:
      - hasDocuments:
          count: 0
  - it: manifest should match snapshot
    set:
      deployment:
        enabled: true
    asserts:
      - matchSnapshot: {}

  - it: labels & annotations should be set
    set:
      common:
        labels:
          test_common: test
        annotations:
          test_common: test
      deployment:
        enabled: true
        labels:
          test_deployment: bar
        annotations:
          test_deployment: bar
      pod:
        labels:
          test_pod: foo
        annotations:
          test_pod: foo
    asserts:
      - isKind:
          of: Deployment
      - isSubset:
          path: metadata.labels
          content:
            test_common: test
            test_deployment: bar
      - isSubset:
          path: metadata.annotations
          content:
            test_common: test
            test_deployment: bar
      - isSubset:
          path: spec.template.metadata.labels
          content:
            test_common: test
            test_pod: foo
      - isSubset:
          path: spec.template.metadata.annotations
          content:
            test_common: test
            test_pod: foo
