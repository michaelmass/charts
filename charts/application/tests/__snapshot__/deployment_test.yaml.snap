manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: application
        helm.sh/chart: application-0.0.1
      name: RELEASE-NAME-application
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: application
      strategy:
        type: RollingUpdate
      template:
        metadata:
          annotations: {}
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: application
            helm.sh/chart: application-0.0.1
        spec:
          affinity:
            nodeAffinity: null
            podAffinity: null
            podAntiAffinity:
              preferredDuringSchedulingIgnoredDuringExecution:
                - podAffinityTerm:
                    labelSelector:
                      matchLabels:
                        app.kubernetes.io/instance: RELEASE-NAME
                        app.kubernetes.io/name: application
                    topologyKey: kubernetes.io/hostname
                  weight: 1
          priorityClassName: null
          securityContext:
            fsGroup: 1001
            fsGroupChangePolicy: Always
            supplementalGroups: []
            sysctls: []
